libnetwork/cnmallocator/provider_test.go

```diff
@@ -0,0 +1,30 @@
+package cnmallocator
+
+import (
+	"testing"
+
+	"github.com/moby/swarmkit/v2/api"
+	"github.com/moby/swarmkit/v2/testutils"
+	"github.com/stretchr/testify/assert"
```

Comment:
Looks like we pulled in testify here for the test (looking at the code, it's probably trivial to rewrite to gotest.tools or plain go?)

Reply:
Probably, [but that wouldn't get testify out of the dependency graph.](
https://github.com/moby/swarmkit/blob/df1e66110903b9bee3e377df5b790024268dc42d/manager/allocator/allocator_test_suite.go#L21-L25)
https://github.com/moby/moby/blob/6fe7928c8c9b29bcd2d0232c356e76134656901f/libnetwork/cnmallocator/allocator_test.go#L1-L11

If moby having a testify dependency is a problem I'll need to rewrite that test suite in Swarmkit.

Reply:
Discussed in maintainer call. Having multiple assertion libraries vendored makes it too easy to import the non-preferred library when writing new tests and have it slip past review. Rewriting these tests to use gotest.tools would not get rid of the testify dependency because of `allocator.RunAllocatorTests`; it would only make testify a transitive dependency. I will update this PR to add a lint rule to ban direct imports of testify packages.

Reply:
@thaJeztah you were right: [it was trivial to rewrite.](https://pkg.go.dev/gotest.tools/v3@v3.5.1/assert/cmd/gty-migrate-from-testify)